src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -2,17 +2,29 @@
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
-import io.stargate.sgv2.jsonapi.api.model.command.clause.update.UpdateClause;
-import io.stargate.sgv2.jsonapi.api.model.command.clause.update.UpdateOperation;
+import io.smallrye.mutiny.Uni;
+import io.stargate.sgv2.jsonapi.api.model.command.clause.update.*;
 import io.stargate.sgv2.jsonapi.config.constants.DocumentConstants;
 import io.stargate.sgv2.jsonapi.exception.ErrorCode;
 import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import io.stargate.sgv2.jsonapi.service.embedding.DataVectorizer;
+import io.stargate.sgv2.jsonapi.service.operation.model.impl.ReadDocument;
 import io.stargate.sgv2.jsonapi.util.JsonUtil;
 import java.util.List;
 
-/** Updates the document read from the database with the updates came as part of the request. */
+/**
+ * Updates the document read from the database with the updates came as part of the request.
+ * DocumentUpdater construct postpone from commandResolver level to operation level, since we want
+ * the vectorize as needed, only vectorize when there are documents found or upsert
+ *
+ * @param updateClause - vectorize it as needed before building update operations
+ * @param replaceDocument - replaceDocument to replace the one read from DB
+ * @param replaceDocumentId - documentId from replaceDocument
+ * @param updateType - UPDATE/REPLACE
+ */
 public record DocumentUpdater(
-    List<UpdateOperation> updateOperations,
+    // buildOperations will be executed when apply to update
+    UpdateClause updateClause,
```

Comment:
Why does this now get the clause and not the operations ? That feels like a big change

Reply:
also, if all the constructions is via static factory methods we should make the ctor private